### PR TITLE
DOC: Docstring for pandas.index.max

### PR DESCRIPTION
**Pull request by JoanMartin**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
######################### Docstring (pandas.Index.max) #########################
################################################################################

Return the maximum value of the object.

Return the maximum value of the object within the same type.

Returns
-------
scalar
    Maximum value.

See Also
--------
Series.max : Return the maximum value of the object.
DataFrame.max : Return the maximum value of the object.
Index.min : Return the minimum value of the object.

Examples
--------
>>> idx = pd.Index([3, 2, 1])
>>> idx.max()
3

>>> idx = pd.Index(['c', 'b', 'a'])
>>> idx.max()
'c'

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Index.max" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -787,7 +787,31 @@ def empty(self):
         return not self.size
 
     def max(self):
-        """ The maximum value of the object """
+        """
+        Return the maximum value of the object.
+
+        Return the maximum value of the object within the same type.
+        Remember you also can access to a Dataframe's index as Dataframe.index.
```
Comment (jcorrius) [inline pandas/core/base.py]: This sentence may need some rewording

File: pandas/core/base.py [hunk 2]
```diff
@@ -791,7 +791,7 @@ def max(self):
         Return the maximum value of the object.
 
         Return the maximum value of the object within the same type.
-        Remember you also can access to a Dataframe's index as Dataframe.index.
+        The Index of a DataFrame can be accessed as pandas.DataFrame.index.
```
Comment (rafarui) [inline pandas/core/base.py]: I would rewrite the last sentence to:
The DataFrame Index can be accessed using pandas.DataFrame.index.

**Comment by caperutxa** [comment]
Looks so nice

File: pandas/core/base.py (hunk 2 above)
Comment (JoanMartin) [reply pandas/core/base.py]: Thanks! :)

File: pandas/core/base.py [hunk 3]
```diff
@@ -787,7 +787,31 @@ def empty(self):
         return not self.size
 
     def max(self):
-        """ The maximum value of the object """
+        """
+        Return the maximum value of the object.
+
+        Return the maximum value of the object within the same type.
+        The DataFrame Index can be accessed using pandas.DataFrame.index.
```
Comment (jreback) [inline pandas/core/base.py]: this last sentence is unecessary

File: pandas/core/base.py [hunk 4]
```diff
@@ -787,7 +787,31 @@ def empty(self):
...
+
+        Returns
+        -------
+        object
```
Comment (jreback) [inline pandas/core/base.py]: scalar

File: pandas/core/base.py [hunk 5]
```diff
@@ -787,7 +787,31 @@ def empty(self):
...
+            Maximum value.
+
+        See Also
+        --------
+        Index.min : Return the minimum value of the object.
+
```
Comment (jreback) [inline pandas/core/base.py]: link to Series.max, DataFrame.max

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/base.py (hunk 5 above)
Comment (JoanMartin) [reply pandas/core/base.py]: Thanks! =)

**Comment by TomAugspurger** [comment]
Thanks @JoanMartin 👍

**Pull request merged**